services/inputhost/pathCache.go

```diff
@@ -508,3 +508,16 @@ func (pathCache *inPathCache) getState() *admin.DestinationState {
 
 	return destState
 }
+
+func (pathCache *inPathCache) drainExtent(extUUID string) error {
+	pathCache.RLock()
+	defer pathCache.RUnlock()
+
+	extCache, ok := pathCache.extentCache[extentUUID(extUUID)]
```

Comment:
You might have already thought through this and may be some of the following things are planned for another diff. But questions for me:
(1) shouldn't there by a drain in pubConnection as well ?
(2) If you have (1), do you still need a drain in the extHost as well  ? 
(3) If you have drain in both pubConn & extHost, don't you have to call pubConn.drain() for all conns, wait before calling extHost.drain() ?